lib/storage/dedup.go

```diff
@@ -34,20 +34,36 @@ func DeduplicateSamples(srcTimestamps []int64, srcValues []float64, dedupInterva
 	tsNext -= tsNext % dedupInterval
 	dstTimestamps := srcTimestamps[:0]
 	dstValues := srcValues[:0]
+
+	var prevTs int64
+	var value, prevValue float64
 	for i, ts := range srcTimestamps[1:] {
+		value = srcValues[i]
+		if i > 0 && srcTimestamps[i] == srcTimestamps[i-1] && value < prevValue {
```

Comment:
I think, that we shouldn't compare timestamps for equality, because de-duplication happen over time interval. Why do not choose biggest value on  dedupInterval?

 For instance, two duplicate series:
series 1:
  ts[0,5,10,15,20], values[1,2,3,4,5]
series 2 :
 ts[0,4,10,14,20] values [1,5,6,9,10].

 With 5 ms dedup interval it's possible to get random values after de-duplication.

Reply:
> Why do not choose biggest value on dedupInterval?

because there is still an option to "update" the recently written value if you send a new value with a bit shifted timestamp within the dedupInterval. The problem appears when you have two identical timestamps and you need to choose. Before, we chose randomly. Now we apply additional logic for values comparing removing the randomness.

Choosing the biggest value on the interval will change the current behavior and will make "mistakes fixing" on already written data impossible.